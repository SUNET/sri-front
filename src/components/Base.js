import React from "react";
import { Route, Switch } from "react-router-dom";

import SearchContainer from "../containers/Search";
import { ContactDetails } from "./Contact";

class Base extends React.Component {
    render() {
        return (
            <main className="container">
                <Switch>
                    <Route path="/contacts" component={SearchContainer} />
                    <Route
                        path="/contact/:contactId"
                        component={ContactDetails}
                    />
                    <Route path="/roles" component={SearchContainer} />
                </Switch>
            </main>
        );
    }
}

export default Base;
