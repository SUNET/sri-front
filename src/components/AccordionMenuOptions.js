import React from "react";
import { Nav } from "react-bootstrap";
import { NavLink } from "react-router-dom";
import ToggleSection, { ToggleHeading, TogglePanel } from "../components/ToggleSection";

import "../style/AccordionMenuOptions.scss";

class AccordionMenuOptions extends React.Component {
    renderHeader(option, index) {
        return (
            <ToggleHeading key={`${index}`}>
                <div className="accordion-menu-options__header">{option.header}</div>
            </ToggleHeading>
        );
    }
    renderCollapseList(subOptions, optionIndex) {
        return (
            <TogglePanel>
                {subOptions.map((subOption, subOpIndex) => {
                    return (
                        <div key={`option_menu-${optionIndex}-${subOpIndex}`} className="accordion-menu-options__link">
                            <Nav.Link
                                as={NavLink}
                                activeClassName="active"
                                to={subOption.route}
                                onClick={() => this.props.onSelectedOption()}
                            >
                                <i className={`accordion-menu-options__link__icon ${subOption.iconClass}`}></i>
                                <span className="accordion-menu-options__link__name">{subOption.name}</span>
                            </Nav.Link>
                        </div>
                    );
                })}
            </TogglePanel>
        );
    }
    render() {
        const { data } = this.props;

        return (
            <div className="accordion-menu-options">
                {data.map((option, optionIndex) => {
                    return (
                        <ToggleSection key={`option_menu-${optionIndex}`}>
                            {this.renderHeader(option, optionIndex)}
                            {this.renderCollapseList(option.subOptions, optionIndex)}
                        </ToggleSection>
                    );
                })}
            </div>
        );
    }
}

export default AccordionMenuOptions;
