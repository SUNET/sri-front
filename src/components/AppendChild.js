import React from 'react';
import { Field } from 'redux-form';
import { withTranslation } from 'react-i18next';
import FieldInput from './FieldInput';

class AppendChild extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      fields: [{ name: '' }],
    };
  }

  componentWillMount() {}

  handleFieldNameChange = (index) => (event) => {
    const newShareholders = this.state.fields.map((field, sindex) => {
      if (index !== sindex) return field;
      return { ...field, name: event.target.value };
    });

    this.setState({ fields: newShareholders });
  };

  handleAddFild = (event) => {
    event.preventDefault();
    this.setState({
      fields: this.state.fields.concat([{ name: '' }]),
    });
  };

  render() {
    let { t } = this.props;
    return (
      <div>
        {this.state.fields.map((field, index) => (
          <Field
            component={FieldInput}
            type="text"
            placeholder={`${field.name}${index + 1}`}
            value={field.name}
            onChange={this.handleFieldNameChange(index)}
          />
        ))}
        <button className="btn btn-add outline" onClick={(e) => this.handleAddField(e)}>
          <span>{t('actions/add-new')}</span>
        </button>
      </div>
    );
  }
}

export default withTranslation()(AppendChild);
