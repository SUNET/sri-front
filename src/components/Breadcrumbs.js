import React from 'react';
import Breadcrumb from 'react-bootstrap/Breadcrumb';
import PropTypes from 'prop-types';
import Routes from '../Routes';

function generatePathsArray(pathString, isDetailsEntity, entityDetails) {
  return pathString.split('/').map((pathElement, index, array) => {
    const mathIdCurrentDetailsInLocationPath = pathString.includes(entityDetails.id);
    const isLastPath = index === array.length - 1;
    const isActive = isLastPath;
    let pathTo = `/${pathElement}`;
    let name = Routes[pathTo];

    if (typeof Routes[pathTo] === 'undefined') {
      pathTo = array.slice(0, index + 1).join('/');
      name = isLastPath && isDetailsEntity && mathIdCurrentDetailsInLocationPath ? entityDetails.name : Routes[pathTo];
    }
    return {
      pathTo,
      name,
      isActive,
    };
  });
}

const Breadcrumbs = ({ history, isDetailsEntity, entityDetails }) => {
  const { pathname } = history.location;
  const routeSteps = generatePathsArray(pathname, isDetailsEntity, entityDetails);
  return (
    <div className="breadcrumbs">
      <Breadcrumb>
        {routeSteps.map(({ pathTo, name, isActive }, index) => {
          return (
            <Breadcrumb.Item active={isActive} key={`${index * Math.random()}`} href={pathTo}>
              {name}
            </Breadcrumb.Item>
          );
        })}
      </Breadcrumb>
    </div>
  );
};

Breadcrumbs.propsTypes = {
  history: PropTypes.object.isRequired,
  isDetailsEntity: PropTypes.bool.isRequired,
  entityDetails: PropTypes.bool.isRequired,
};

export default Breadcrumbs;
