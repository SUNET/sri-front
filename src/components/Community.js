import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { withTranslation } from "react-i18next";

import SearchOrganizationContainer from "../containers/organization/SearchOrganization";
import SearchContactContainer from "../containers/contact/SearchContact";
import SearchGroupContainer from "../containers/group/SearchGroup";
// import SubMenu from "./SubMenu";
// import SubMenuActions from "./SubMenuActions";

import "../style/Footer.scss";

class Community extends React.Component {
    render() {
        return (
            <Switch>
                <Redirect exact from="/community" to="/community/organizations" />
                <Route path="/community/organizations" component={SearchOrganizationContainer} />
                <Route path="/community/contacts" component={SearchContactContainer} />
                <Route path="/community/groups" component={SearchGroupContainer} />
            </Switch>
        );
    }
}

export default withTranslation()(Community);
