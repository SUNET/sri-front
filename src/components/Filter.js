import React from "react";
import { Form } from "react-bootstrap";
import { withTranslation } from "react-i18next";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFilter } from "@fortawesome/free-solid-svg-icons";

import Dropdown from "./Dropdown";

import "../style/Filter.scss";

class Filter extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            filterValue: "",
            contact_type: ""
        };
    }

    render() {
        const t = this.props.t;
        return (
            <section>
                <Form className="filter">
                    <div>
                        <Form.Control
                            placeholder="Filter"
                            defaultValue={this.state.filterValue}
                            onChange={(e) => this.props.handleOnChangeFilter(e)}
                        />
                    </div>
                    <div className="mt-1">
                        <header>
                            <span>
                                {t("Filters")}
                            </span>
                            <FontAwesomeIcon
                                icon={faFilter}
                                className="pullRight"
                            />
                        </header>
                        <Dropdown
                            label="Role"
                            type="contact_type"
                            onChange={(e) =>
                                this.setState({
                                    contact_type: e.target.value
                                })
                            }
                        />
                    </div>
                </Form>
            </section>
        );
    }
}

export default withTranslation()(Filter);
