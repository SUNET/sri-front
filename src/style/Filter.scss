@import "../variables.scss";

.filter {
    position: relative;
    margin-right: 16px;
    display: inline-block;

    &:before {
        display: inline-block;
        content: "\e90c";
        font-family: "sunet-icons";
        color: $complementary-color;
        position: absolute;
        font-size: 20px;
        left: 14px;
        top: 4px;
    }

    input {
        width: 204px;
        border-radius: $input--border-radius;
        border-width: 1px;
        height: $input--height;
        padding: 0 10px 0 40px;
        font-size: 14px;
    }

    .clear-input {
        position: absolute;
        right: 14px;
        top: 9px;
        color: $secondary-color-30;
        cursor: pointer;

        &:hover {
            color: $complementary-color;
        }
    }
}

.filter-date {
    .pretty {
        margin-right: 24px;

        label {
            text-indent: 16px !important;
            font-family: Karla;

            &:after,
            &:before {
                top: calc((0% - (100% - 18px)) - 8%) !important;
                font-size: 10px;
            }
        }
    }
}
