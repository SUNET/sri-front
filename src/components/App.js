import React, { Component } from "react";
import PropTypes from "prop-types";
import { ConnectedRouter } from "connected-react-router";
import Breadcrumbs from "react-router-dynamic-breadcrumbs";
import { Route, Switch } from "react-router-dom";

import FetchingContext from "../components/FetchingContext";
import SplashContainer from "../containers/Splash";
import NotifyContainer from "../containers/Notify";
import TopHeaderContainer from "../containers/TopHeader";
import BaseContainer from "../containers/Base";
import FooterContainer from "../containers/Footer";

import { Row, Col } from "react-bootstrap";
import SideNavNetwork from "./SideNavNetwork";
import SideNavCommunity from "./SideNavCommunity";
import SubMenuActions from "./SubMenuActions";

import Routes from "../Routes";

import "bootstrap/scss/bootstrap.scss";
import "../style/App.scss";
import "../style/Breadcrumbs.scss";
import "../style/SRIButton.scss";

import { history } from "../store";

class App extends Component {
    render() {
        return (
            <FetchingContext.Provider value={this.props.is_fetching}>
                <div className="App container">
                    <ConnectedRouter history={history}>
                        <Row>
                            <Col className="px-0">
                                <TopHeaderContainer />
                            </Col>
                        </Row>
                        <Row>
                            <SplashContainer />
                            <Switch>
                                <Route path="/network" component={SideNavNetwork} />
                                <Route path="/community" component={SideNavCommunity} />
                            </Switch>
                            <Col sm={10} className="fixed-adaptative">
                                <NotifyContainer />
                                <Row className="mt-4">
                                    <Col>
                                        <Breadcrumbs mappedRoutes={Routes} />
                                    </Col>
                                    <Col className="text-right">
                                        <SubMenuActions />
                                    </Col>
                                </Row>
                                {this.props.is_app_loaded && <BaseContainer />}
                            </Col>
                        </Row>
                        <Row>
                            <Col className="px-0">
                                <FooterContainer />
                            </Col>
                        </Row>
                    </ConnectedRouter>
                </div>
            </FetchingContext.Provider>
        );
    }
}

App.propTypes = {
    is_fetching: PropTypes.bool,
    is_app_loaded: PropTypes.bool,
};

export default App;
