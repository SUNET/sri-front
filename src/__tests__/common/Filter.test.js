import React from 'react';
import { mount } from 'enzyme';
import { Filter } from '../../components/Filter';
import toJson from 'enzyme-to-json';

let FilterWrapper;
let FilterComponent;
let FilterComponentInstance;
let InputElement;

const defaultProps = {
  changeFilter: jest.fn(),
  t: jest.fn((t) => t),
};

const CreateFilterWrapper = (props) => {
  const allProps = { ...defaultProps, ...props };
  return (
    <div>
      <Filter {...allProps} />
    </div>
  );
};

const mountFilterWrapper = (props) => mount(<CreateFilterWrapper {...props} />);
beforeEach(() => {
  jest.clearAllMocks();
  FilterWrapper = mountFilterWrapper();
  FilterComponent = FilterWrapper.find('Filter');
  FilterComponentInstance = FilterComponent.instance();
  FilterComponentInstance.changeFilterDebounce = FilterComponentInstance.changeFilter;
  InputElement = FilterComponent.find('input');
});
afterEach(() => {
  FilterWrapper.unmount();
});

describe('Filter Component', () => {
  it('Renders correctly', () => {
    expect(toJson(FilterWrapper)).toMatchSnapshot();
    expect(FilterComponent.find('.filter')).toHaveLength(1);
  });
});
describe('Filter Elements', () => {
  it('Remove text icon: No appears when input is clean', () => {
    expect(FilterComponent.state().filterValue).toBe('');
    expect(FilterWrapper.find('.clear-input-cta')).toHaveLength(0);
  });
  it('Remove text icon: Appears when input is filled', () => {
    InputElement.simulate('change', { target: { value: 'new search' } });
    expect(FilterComponent.state().filterValue).toBe('new search');
    expect(FilterWrapper.find('.clear-input-cta')).toHaveLength(1);
  });
});
describe('Clear Input', () => {
  it('The input is cleared when the X is pressed when there is written text', () => {
    InputElement.simulate('change', { target: { value: 'text to be deleted' } });
    const cleanCTA = FilterWrapper.find('.clear-input-cta');
    expect(FilterComponent.state().filterValue).toBe('text to be deleted');
    cleanCTA.simulate('click');
    expect(FilterComponent.state().filterValue).toBe('');
    expect(FilterWrapper.find('.clear-input-cta')).toHaveLength(0);
  });
});
describe('The search method is called', () => {
  it('When you write in the input, the props search method is called', () => {
    InputElement.simulate('change', { target: { value: 'text to search' } });
    expect(defaultProps.changeFilter).toHaveBeenCalledWith('text to search');
    expect(defaultProps.changeFilter).toHaveBeenCalledTimes(1);
  });
});
