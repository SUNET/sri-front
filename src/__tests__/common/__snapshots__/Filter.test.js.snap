// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Renders correctly 1`] = `
<CreateFilterWrapper>
  <div>
    <Filter
      changeFilter={[MockFunction]}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "filter.text.placeholder",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "filter.text.placeholder",
            },
          ],
        }
      }
    >
      <div
        className="filter"
      >
        <FormControl
          name="filter"
          onChange={[Function]}
          placeholder="filter.text.placeholder"
          type="text"
          value=""
        >
          <input
            className="form-control"
            name="filter"
            onChange={[Function]}
            placeholder="filter.text.placeholder"
            type="text"
            value=""
          />
        </FormControl>
      </div>
    </Filter>
  </div>
</CreateFilterWrapper>
`;
